projects/hr-portal/src/hr-portal.ts

```diff
@@ -1,83 +1,195 @@
+configureTheme("fluent");
 
 @customElement("app-hr-portal")
 export default class HrPortal extends LitElement {
-  render() {
-    return html` <div>HR PORTAL</div> `;
+
+  @property({ type: Array })
+  data: Array<any> = [];
+
+  constructor() {
+    super();
+    this.data = [];
+    icons.forEach((icon: { name: string; path: string; category: string }) => {
+      registerIcon(icon.name, icon.path, icon.category);
+    });
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.fetchData();
+  }
+
+  async fetchData() {
+    try {
+      const response = await fetch('https://staging7.infragistics.com/grid-examples-data/data/hr/hr.json');
+      const jsonData = await response.json();
+      console.log('Fetched data:', jsonData); // Log the fetched data
+      this.data = jsonData; 
+      console.log('Assigned data:', this.data); // Log the assigned data
+    } catch (error) {
+      console.error('Error fetching HR data:', error);
+    }
+  }
+
+  clearSorting() {
+    const treeGrid = this.shadowRoot?.getElementById('treeGrid') as IgcTreeGridComponent;
+    if (treeGrid) {
+      treeGrid.sortingExpressions = [];
+    }
+  }
+  
+  public avatarTemplate = (ctx: IgcCellTemplateContext) => {	
+    let row = ctx.cell.row;
+    return html`	
+    <div class="employeeDiv">
+      <igc-avatar shape="rounded" .src="${row.data.Picture}"></igc-avatar>
+      <span>${row.data.Name}</span>
+    </div>
+    `;
+  }
+
+  public countryIconTemplate = (ctx: IgcCellTemplateContext) => {
+    let row = ctx.cell.row;
+    return html`
+            <div class="flagDiv">
+              <igc-icon collection="country-icons" name="${row.data.Country}"></igc-icon>
+              <span>${row.data.Location}, ${row.data.Country}</span>
+            </div>`;
+  }
+
+  public contactsTemplate = (ctx: IgcCellTemplateContext) => {
+    let row = ctx.cell.row;
+    return html`
+            <div class="center-content">
+              <a href="mailto:${row.data.Email}">
+                <igc-icon-button collection="hr-icons" name="mail" class="small" variant="flat"></igc-icon-button>
+              </a>
+              <a href="tel:${row.data.Phone}">
+                <igc-icon-button collection="hr-icons" name="tel" class="small" variant="flat"></igc-icon-button>
+              </a>
+              <a href="https://www.linkedin.com" target="_blank">
+                <igc-icon-button collection="hr-icons" name="linkedIn" class="small" variant="flat"></igc-icon-button>
+              </a>
+            </div>
+          `;
   }
 
+  public dateTemplate = (ctx: IgcCellTemplateContext) => {
+    let row = ctx.cell.row;
+    const date = new Date(row.data.HireDate);
+    const formattedDate = date.toISOString().split('T')[0]; 
+    return html`
+      ${formattedDate}
+    `;
+  }
+
+  render() {
+    return html`
+      <link rel="stylesheet" href="node_modules/igniteui-webcomponents-grids/grids/themes/light/fluent.css" />
+      <igc-tree-grid
```

Comment:
There is a tree grid selector that is redundant

Reply:
removed